wled00/FX.cpp

```diff
@@ -4861,6 +4861,53 @@ static const char _data_FX_MODE_FLOWSTRIPE[] PROGMEM = "Flow Stripe@Hue speed,Ef
 #define XY(x,y) SEGMENT.XY(x,y)
 
 
+/////////////////////////////
+//    2D Analog Clock      //
+/////////////////////////////
+uint16_t mode_2DAnalogClock(void) {                  // By Andras Fekete (bandi13)
+  if (!strip.isMatrix || !SEGMENT.is2D()) return mode_static(); // not a 2D set-up
+
+  const int cols = SEGMENT.virtualWidth();
+  const int rows = SEGMENT.virtualHeight();
+  const int centerX = (cols-!(cols%2)) >> 1;  // use odd sized circle
+  const int centerY = (rows-!(rows%2)) >> 1;  // use odd sized circle
+  const int radius = min(centerX, centerY);
+  const bool soft = radius > 6 && SEGMENT.check2;
+
+  SEGMENT.fadeToBlackBy(240);
+  SEGMENT.drawCircle(centerX, centerY, radius, DARKGREY, soft);
+  if (radius > 8) SEGMENT.drawCircle(centerX, centerY, radius-1, DARKGREY, soft); // thicker circle
+
+  time_t hours_in_day = (localTime / (60 * 60)) % 12;
+  time_t minutes_in_day = (localTime / (60)) % 60;
+  time_t seconds_in_day = (localTime) % 60;
+
+  float hour_angle = radians(30.0f * (hours_in_day + minutes_in_day / 60.0f) - 90.0f);
+  int hour_len = radius / 2;
+  int hour_x = centerX + hour_len * cos_t(hour_angle);
+  int hour_y = centerY + hour_len * sin_t(hour_angle);
+
+  float minute_angle = radians(6.0f * (minutes_in_day + seconds_in_day / 60.0f) - 90.0f);
+  int minute_len = (radius * 7) / 10;
+  int minute_x = centerX + minute_len * cos_t(minute_angle);
+  int minute_y = centerY + minute_len * sin_t(minute_angle);
+
+  if (SEGMENT.check1) {
+    float second_angle = radians(6.0f * seconds_in_day - 90.0f);
+    int second_len = (radius * 9) / 10;
+    int second_x = centerX + second_len * cos_t(second_angle);
+    int second_y = centerY + second_len * sin_t(second_angle);
+    SEGMENT.drawLine(centerX, centerY, second_x, second_y, SEGCOLOR(2), soft);
+  }
+
+  SEGMENT.drawLine(centerX, centerY, minute_x, minute_y, SEGCOLOR(1), soft);
+  SEGMENT.drawLine(centerX, centerY, hour_x, hour_y, SEGCOLOR(0), soft);
+
+  return FRAMETIME;
```

Comment:
Why not calculate this once per second? No need to do it any more or less.

Reply:
Agreed that there should be no need to execute it more (or less) often than every 1000ms. But default FPS is 42 which does not divide 1000 nicely (without remainder). So there will be situations where seconds hand will behave erratic if you'd return 1000 (that is the amount of time needed until next execution) and not `1000 / FPS` (which is FRAMETIME when using default FPS).

Try it on small dimension. And with TM1814 LEDs.
Each effect should return FRAMETIME to allow seamless transitions. The return value is deprecated and is kept for backwards compatibility until all effects are rewritten to be executed at the user specified FPS.